### PR TITLE
feat:support zstd compress and uncompressed

### PR DESCRIPTION
**Pull request by Max-Cheng**

### CODE REVIEW

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]
Tests are failing.

**Comment by Max-Cheng** [comment]
> Tests are failing.

Cause [compress](https://github.com/klauspost/compress) is declaring `go.mod' with `go 1.19'. In this case, I'll downgrade the version.

**Comment by erikdubbelboer** [comment]
Still failing.

**Comment by Max-Cheng** [comment]
> Still failing.

Ok. Looks like 1.18 `encoding/binary` does not implement `AppendByteOrder`.
[1.18](https://cs.opensource.google/go/go/+/release-branch.go1.18:src/encoding/binary/binary.go)
[1.19](https://cs.opensource.google/go/go/+/release-branch.go1.19:src/encoding/binary/binary.go;drc=986b04c0f12efa1c57293f147a9e734ec71f0363;l=46)

I will fix this in other ways.

**Comment by gaby** [comment]
Compress minimum Go version is 1.19, downgrading is not a good idea since it will *not* have the corruption fixes made to `zstd` recently.

@erikdubbelboer Any timeline for fasthttp to bump the min go version?

**Comment by erikdubbelboer** [comment]
I'm ok with you dropping 1.18 support in this pull. It's not officially maintained by the Go team either.

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,8 @@ module github.com/valyala/fasthttp
 go 1.20
 
 require (
-	github.com/andybalholm/brotli v1.0.5
-	github.com/klauspost/compress v1.17.0
+	github.com/andybalholm/brotli v1.1.0
+	github.com/klauspost/compress v1.17.2
```
Comment (gaby) [inline go.mod:7]: This should be v1.17.5

**Review by gaby — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (gaby) [reply go.mod:7]: v1.17.6 was released today

Comment (Max-Cheng) [reply go.mod:7]: Glad to see that, I'll update later.

File: zstd.go [hunk 2]
```diff
@@ -150,7 +151,7 @@ func nonblockingWriteZstd(ctxv any) {
 
 // AppendZstdBytes appends zstd src to dst and returns the resulting dst.
 func AppendZstdBytes(dst, src []byte) []byte {
-	return AppendZstdBytesLevel(dst, src, CompressBrotliDefaultCompression)
+	return AppendZstdBytesLevel(dst, src, CompressZstdDefault)
```
Comment (gaby) [inline zstd.go:154]: Good catch!

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]
There are still linting errors and it needs to be rebased on master now.

**Comment by Max-Cheng** [comment]
Please Hold. I'll fix this error by rebase error

**Comment by Max-Cheng** [comment]
@erikdubbelboer fixed

**Comment by gaby** [comment]
@erikdubbelboer The timeouts in the tests are too agressive. The same error happens in a different PR

https://github.com/valyala/fasthttp/actions/runs/7881870997/job/21506194694?pr=1720

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go-version: [1.18.x, 1.19.x, 1.20.x, 1.21.x, 1.22.x]
+        go-version: [1.19.x, 1.20.x, 1.21.x]
```
Comment (alexandear) [inline .github/workflows/test.yml:12]: I'm curious, do we need to change supported versions in this PR?

**Comment by gaby** [comment]
@alexandear @erikdubbelboer @Max-Cheng Can someone rebase with master to see what's left on this PR?

**Comment by gaby** [comment]
@erikdubbelboer If you add this to the lint workflow before the job, it will show you here on Github where the issues are:

```yaml
permissions:
  # Required: allow read access to the content for analysis.
  contents: read
  # Optional: allow read access to pull request. Use with `only-new-issues` option.
  pull-requests: read
  # Optional: Allow write access to checks to allow the action to annotate code in the PR.
  checks: write
```

Example: https://github.com/gofiber/fiber/blob/main/.github/workflows/linter.yml

**Comment by erikdubbelboer** [comment]
@gaby thanks for the suggestion. I think everything is fixed now and this can be merged.

**Pull request merged**